runbotics/runbotics-orchestrator-ui/src/src-app/views/process/EditProcessDialog.tsx

```diff
@@ -29,71 +33,147 @@ type EditProcessDialogProps = {
     onAdd: (process: IProcess) => void;
 };
 
+export interface FormValidationState {
+    name: boolean;
+}
+
+const initialFormValidationState: FormValidationState = {
+    name: true
+};
+
+const getProcessDataWithoutNulls = (process: IProcess): IProcess => ({
+    ...process,
+    description: process.description ?? ''
+});
+
+const getProcessDataWithoutEmptyStrings = (process: IProcess): IProcess => ({
+    ...process,
+    description: (process.description === '' ? null : process.description)
+});
+
 const EditProcessDialog: FC<EditProcessDialogProps> = ({
     process, onAdd, onClose, open,
 }) => {
-    const [formState, setFormState] = useState<IProcess>({ ...process });
     const { translate } = useTranslations();
+    const dispatch = useDispatch();
 
-    const handleSubmit = () => {
-        onAdd(formState);
+    const [formValidationState, setFormValidationState] = useState<FormValidationState>(initialFormValidationState);
+    const [processFormState, setProcessFormState] = useState<IProcess>();
+    const [searchedDatabaseTags, setSearchedDatabaseTags] = useState<Tag[]>([]);
+    const [selectedTagsNames, setSelectedTagsNames] = useState<string[]>([]);
+    const [search, setSearch] = useState<string>('');
+    const debouncedSearch = useDebounce<string>(search.trim(), DEBOUNCE_TIME);
+
+    const handleTagChange = (updatedTags: string[]) => {
+        setSelectedTagsNames(updatedTags);
+
+        const hasTagBeenAdded = updatedTags.length > selectedTagsNames.length;
+        if (hasTagBeenAdded) {
+            const tagToAdd = (searchedDatabaseTags.find(tag => tag.name === updatedTags.at(-1))) ?? { name: updatedTags.at(-1) };
+            setProcessFormState((prevState) => ({
+                ...prevState,
+                tags: [...processFormState.tags, tagToAdd]
+            }));
+        } else {
+            const filteredTags = [...processFormState.tags].filter((tag) => updatedTags.includes(tag.name));
+            setProcessFormState((prevState) => ({
+                ...prevState,
+                tags: filteredTags
+            }));
+        }
+
+        setSearch('');
+    };
+
+    const handleTagSearch = (e: ChangeEvent<HTMLInputElement>) => {
+        const filteredSearch = e.target.value.replaceAll(/[^A-Za-z0-9]/g, '');
+        if (filteredSearch.length <= MAX_TAG_LENGTH) setSearch(filteredSearch);
     };
 
-    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-        setFormState({ ...formState, [e.target.name]: e.target.value });
+    const handleAddByEnter = (e: KeyboardEvent<HTMLInputElement>) => {
+        if(e.key === 'Enter'
+            && search !== ''
+            && selectedTagsNames.length < MAX_NUMBER_OF_TAGS
```

Comment:
so we cannot add more tags than there is place for displaying them? I know there are limitations on the DB side but what if they change there to less strict?

Reply:
Tag section while expanded is hiding all process details, because it looks clearer. There will be always more space than needed.